src/targets/browser/browserLauncher.ts

```diff
@@ -215,16 +279,21 @@ export class BrowserLauncher implements ILauncher {
       this._onTargetListChangedEmitter.fire();
     });
 
-    const filter =
-      params.useWebView === 'advanced' ? createTargetFilterForConfig(params) : undefined;
+    const filterForMain = params.useWebView ? filter : undefined;
 
     // Note: assuming first page is our main target breaks multiple debugging sessions
     // sharing the browser instance. This can be fixed.
-    this._mainTarget = await timeoutPromise(
-      this._targetManager.waitForMainTarget(filter),
-      cancellationToken,
-      'Could not attach to main target',
-    );
+    const mainTargetPromise = this._targetManager.waitForMainTarget(filterForMain);
+
+    if (params.useWebView === 'advanced') {
+      this._mainTarget = await mainTargetPromise;
+    } else {
+      this._mainTarget = await timeoutPromise(
+        mainTargetPromise,
+        cancellationToken,
+        'Could not attach to main target',
+      );
```

Comment:
I'm not sure this needs to be a separate codepath. People using webview, or the EdgeConfigurationProvider (once made) could just set this timeout to an arbitrarily high number. Or maybe we support `0` to be a never cancelled timeout, at the global level.

Reply:
That sounds good to me, I just pushed a commit that removes the timeout-free code path.

I originally added the timeout-free code path for advanced web view debugging to match vscode-edge-debug2 behavior but prefer this with the instruction to users to set the timeout value for now.

FYI @jalissia

Reply:
I'll defer to you about whether removing the default timeout is generally desirable. If it is then I agree that it should be the default/not require user instruction, and be applied in the configuration resolver 🙂